controller/statsyncer/statsyncer.go

```diff
@@ -0,0 +1,243 @@
+package statsyncer
+
+import (
+	"fmt"
+	"time"
+
+	clusterv1 "github.com/rancher/types/apis/cluster.cattle.io/v1"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const (
+	syncInterval = 5 * time.Second
+)
+
+type StatSyncer struct {
+	clusterName  string
+	Clusters     clusterv1.ClusterInterface
+	ClusterNodes clusterv1.ClusterNodeInterface
+	Pods         corev1.PodInterface
+	Nodes        corev1.NodeInterface
+}
+
+func Register(workload *config.WorkloadContext) {
+	s := &StatSyncer{
+		clusterName:  workload.ClusterName,
+		Clusters:     workload.Cluster.Cluster.Clusters(""),
+		ClusterNodes: workload.Cluster.Cluster.ClusterNodes(""),
+		Pods:         workload.K8sClient.CoreV1().Pods(""),
+		Nodes:        workload.K8sClient.CoreV1().Nodes(),
+	}
+
+	go s.syncResources(syncInterval)
+}
+
+func (s *StatSyncer) syncResources(syncInterval time.Duration) {
+	for range time.Tick(syncInterval) {
+		err := s.updateClusterNodeResources()
+		logrus.Info("Syncing allocated resources")
+		if err != nil {
+			logrus.Warn(err)
+		}
+	}
+}
+
+func (s *StatSyncer) updateClusterNodeResources() error {
+	cluster, err := s.getCluster()
+	if err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
May be we can extract `clusterExists` to the helper methods along with isClusterProvisioned